iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/SV.java

```diff
@@ -84,132 +74,341 @@ public double getA() {
-        return otherSideU(dl.getR(), dl.getX(), dl.getG() / 2.0, dl.getB() / 2.0, 1);
+        return otherSideU(dl.getR(), dl.getX(), dl.getG(), dl.getB(), 0.0, 0.0, 1.0, 0.0);
+    }
+
+    public double otherSideU(DanglingLine dl, boolean splitShuntAdmittance) {
+        if (splitShuntAdmittance) {
+            return otherSideU(dl.getR(), dl.getX(), dl.getG() * 0.5, dl.getB() * 0.5, dl.getG() * 0.5, dl.getB() * 0.5, 1.0, 0.0);
+        } else {
+            return otherSideU(dl.getR(), dl.getX(), dl.getG(), dl.getB(), 0.0, 0.0, 1.0, 0.0);
+        }
+    }
+
+    // Has been modified
+    // Before: otherSideU(hl.getR(), hl.getX(), hl.getG1(), hl.getB1(), 0.0, 0.0, 1.0, 0.0);
+    public double otherSideU(TieLine.HalfLine hl) {
+        return otherSideU(hl.getR(), hl.getX(), hl.getG1(), hl.getB1(), hl.getG2(), hl.getB2(), 1.0, 0.0);
     }
 
+    /**
+     * @deprecated Not used anymore. Use
+     * {@link SV#otherSideA(double, double, double, double, double, double, double, double)}
+     */
+    @Deprecated
     public double otherSideA(double r, double x, double g1, double b1, double rho) {
-        Complex z = new Complex(r, x); // z=r+jx
-        Complex y1 = new Complex(g1, b1); // y1=g1+jb1
-        Complex u2 = computeU2(y1, new Complex(0, 0), z, rho);
-        return Math.toDegrees(u2.getArgument());
+        throw new PowsyblException("Deprecated. Not used anymore");
     }
 
-    public double otherSideA(TieLine.HalfLine hl) {
-        return otherSideA(hl.getR(), hl.getX(), hl.getG1(), hl.getB1(), 1.0);
+    public double otherSideA(double r, double x, double g1, double b1, double g2, double b2, double rho, double alpha) {
+        LinkData.BranchAdmittanceMatrix adm = LinkData.calculateBranchAdmittance(r, x, 1 / rho, -alpha, 1.0, 0.0,
+            new Complex(g1, b1), new Complex(g2, b2));
+        return otherSideA(adm);
     }
 
+    // Has been modified
+    // Before: otherSideA(dl.getR(), dl.getX(), dl.getG() / 2.0, dl.getB() / 2.0, 0.0, 0.0, 1.0, 0.0);
     public double otherSideA(DanglingLine dl) {
-        return otherSideA(dl.getR(), dl.getX(), dl.getG() / 2.0, dl.getB() / 2.0, 1.0);
+        return otherSideA(dl.getR(), dl.getX(), dl.getG(), dl.getB(), 0.0, 0.0, 1.0, 0.0);
+    }
+
+    public double otherSideA(DanglingLine dl, boolean splitShuntAdmittance) {
+        if (splitShuntAdmittance) {
+            return otherSideA(dl.getR(), dl.getX(), dl.getG() * 0.5, dl.getB() * 0.5, dl.getG() * 0.5, dl.getB() * 0.5, 1.0, 0.0);
+        } else {
+            return otherSideA(dl.getR(), dl.getX(), dl.getG(), dl.getB(), 0.0, 0.0, 1.0, 0.0);
+        }
+    }
+
+    // Has been modified
+    // Before: otherSideA(hl.getR(), hl.getX(), hl.getG1(), hl.getB1(), 0.0, 0.0, 1.0, 0.0);
+    public double otherSideA(TieLine.HalfLine hl) {
+        return otherSideA(hl.getR(), hl.getX(), hl.getG1(), hl.getB1(), hl.getG2(), hl.getB2(), 1.0, 0.0);
+    }
+
+    public static double getRho(TwoWindingsTransformer twt) {
```

Comment:
Consider making these functions (`getRho`, `getAlpha`, `getR`, `getX`, ...) private instead of public. They are only helpers to obtain actual parameters of transformers, and are used only in this class.

Reply:
Done